RFCs/FS-1128-better-interpolated-triple-quoted-strings.md

```diff
@@ -21,99 +22,117 @@ let json = $$"""
+
+// It is also allowed to have {, {{, } or }} adjacent to the interpolation delimiters like so:
+$$$"""{{{{41+1}}}} = {42}"""
+// val it: string = "{42} = {42}"
 """
 ```
 
 # Drawbacks
 
-Even though this only extends already existing syntax, it still can be considered yet another way of doing string interpolation. Moreover, it can't be fully aligned with analogous C# feature (raw string literals) and stay backward compatible at the same time, so this could be a source of confusion (but note that triple quoted string literals are already similar but not really aligned with C#'s raw string literals anyway).
+Even though this only extends already existing syntax, it still can be considered yet another way of doing string interpolation.
+
+Moreover, as mentioned in [Detailed design section](#detailed-design) it can't be fully aligned with analogous C# feature (raw string literals - [docs](https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-11#raw-string-literals)) and stay backward compatible at the same time, so this could be a source of confusion
+(note that triple quoted string literals are already not fully aligned with C#'s raw string literals anyway - see [detailed spec for raw strings](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-11.0/raw-string-literal.md)).
 
 # Alternatives
 
 Alternatives:
+
 1. Try to fully align with C#'s raw string literals at the cost of backward compatiblity.
 2. Introduce alternative syntax for delimiting expression fills that does not clash with `{` and `}` which are common in JSON and certain other contexts.
 
 # Compatibility
 
 Please address all necessary compatibility questions:
 
-* Is this a breaking change?
+- Is this a breaking change?
 *No*
-* What happens when previous versions of the F# compiler encounter this design addition as source code?
+- What happens when previous versions of the F# compiler encounter this design addition as source code?
 *A compiler error*
-* What happens when previous versions of the F# compiler encounter this design addition in compiled binaries?
-*AFAICT it should work, changes will hopefully be limited to lexing stage*
-* If this is a change or extension to FSharp.Core, what happens when previous versions of the F# compiler encounter this construct?
+- What happens when previous versions of the F# compiler encounter this design addition in compiled binaries?
+*It should work, changes will hopefully be limited to lexing stage*
+- If this is a change or extension to FSharp.Core, what happens when previous versions of the F# compiler encounter this construct?
 *It is not a change/extension to FSharp.Core*
 
 # Pragmatics
 
 ## Diagnostics
 
-* New compiler error for interpolated string literals with too many consecutive `{` or `}` characters.
-* Otherwise, same compiler errors as for regular (single-`$`) interpolated strings.
+- New compiler error for interpolated string literals with too many consecutive `{` or `}` characters.
```

Comment:
This reminds me, we should probably add a line or two above in the description about those `N * 2 - 1` rules. The question I have about it, does it have to be balanced? If it does have to be, we should have a specific error for that. If it doesn't, we should say so:

```f#
let x = $$$"""is this {{{{{ 42 + 1 }}} legal?"""
```

Does this produce the string `"is this {{42 legal"`? Or does it raise an error?

Reply:
Also, can we already take a stab at what the new error would look like (error number tbd)?

Reply:
No, the extra braces are just string content, so they don't have to be balanced. I'll add a note.

Reply:
Added new error example and a note about (not) balancing extra braces

Reply:
Perfect, thanks!